emails/tests/mgmt_process_emails_from_sqs_tests.py

```diff
@@ -125,26 +138,91 @@ def make_client_error(message="Unknown", code="Unknown", operation_name="Unknown
     return ClientError(err_response, operation_name)
 
 
-def test_process_queue_no_messages():
-    """process_queue can exit after the max time and processing no messages."""
-    res = Command(queue=fake_queue(), max_seconds=3).process_queue()
-    assert res == {
+def log_extra(log_record):
+    """Reconstruct the "extra" argument to the log call"""
+    omit_log_record_keys = set(
+        (
+            "args",
+            "created",
+            "exc_info",
+            "exc_text",
+            "filename",
+            "funcName",
+            "levelname",
+            "levelno",
+            "lineno",
+            "message",
+            "module",
+            "msecs",
+            "msg",
+            "name",
+            "pathname",
+            "process",
+            "processName",
+            "relativeCreated",
+            "stack_info",
+            "thread",
+            "threadName",
+        )
+    )
+    return {
+        key: val
+        for key, val in log_record.__dict__.items()
+        if key not in omit_log_record_keys
+    }
+
+
+def summary_from_exit_log(caplog_fixture):
+    """Get the extra data from the final log message"""
+    last_log = caplog_fixture.records[-1]
+    assert last_log.message == f"Exiting {COMMAND_NAME}"
+    return log_extra(last_log)
+
+
+def test_no_messages(caplog, test_settings):
```

Comment:
question (non-blocking): This test looks like it uses `log_extra` while the other tests use `summary_from_exit_log`. Can this test be written more concisely by using `summary_from_exit_log` too?

Reply:
👍 good idea! Then `log_extra` can go away entirely...

BTW, I'm aiming for the day when I can replace all this with `structlog`:

https://www.structlog.org/en/stable/testing.html

Reply:
Oh, I remember why. This is the only test that examines logs that aren't the final "Exiting process_emails_from_sqs_tests" message.

``log_extra`` takes a ``LogRecord``, a standard Python logging object, and strips out the items that weren't added by ``extra={}``

``summary_from_exit_log`` takes the ``caplog`` item, gets the last ``.record``, asserts that it is an ``Exiting... `` log entry, and calls ``log_extra`` on it.

So, ``test_no_messages`` needs access to ``log_extra``, because it is the one place that is looking at other log entries.

For completeness, I think I'll add a test to ``test_one_message`` that looks at the message processing log entry.